packages/mitex/specs/README.md

```diff
@@ -0,0 +1,314 @@
+```
+
+**Arguments:**
+- num (int): The number of arguments as environment options for the environment.
+- alias (str): Alias command for typst handler.
+  For example, alias `\begin{pmatrix}` to typst's `pmatrix`, as the key in mitex-scope.
+- handle (function): The handler function, as the value of alias in mitex-scope.
+  It receives fixed number of named arguments as environment options,
+  for example `pmatrix(arg0: ..)` or `pmatrix(arg0: .., arg1: ..)`.
+  And it receives variable length arguments as environment body,
+  for matrix environment, it just like the arguments for `mat(1,2; 3, 4)` in equation mode,
+  That is, to receive a two-dimensional array,
+  like `pmatrtix((1, 2,), (3, 4,))` in script mode.
+  Therefore you need to use `(.. arg) = > {..}` to receive them.
+
+**Return:** A spec item and a scope item (none for no scope item)
+
+
+### `sym`
+
+Define a symbol without alias and without handler function, like \alpha => alpha
+
+```typst
+#let sym = ((kind: "sym"), none)
+```
+
+**Return:** A spec item and no scope item (none for no scope item)
+
+
+### `of-sym`
+
+```typst
+#let of-sym(handle) = ((kind: "sym"), (handle: handle))
+```
+
+Define a symbol without alias and with handler function,
+like \negthinspace => h(-(3/18) * 1em)
+
+**Arguments:**
+
+- handle (function): The handler function, as the value of alias in mitex-scope.
+  For example, define `negthinspace` to handle `h(-(3/18) * 1em)` in mitex-scope
+
+**Return:** A symbol spec and a scope item
+
+
+### `left1-op`
+
+Define a left1-op command without handler, like `\limits` for `\sum\limits`
+
+```typst
+#let left1-op(alias) = ((kind: "cmd", args: ( kind: "left1" ), alias: alias), none)
+```
+
+**Arguments:**
+
+- alias (str): Alias command for typst handler.
+  For example, alias `\limits` to typst's `limits`
+  and alias `\nolimits` to typst's `scripts`
+
+**Return:** A cmd spec and no scope item (none for no scope item)
+
+
+### `cmd1`
+
+Define a cmd1 command like \hat{x} => hat(x)
+
+```typst
+#let cmd1 = ((kind: "cmd1"), none)
+```
+
+**Return:** A cmd1 spec and a scope item (none for no scope item)
+
+
+### `cmd2`
+
+Define a cmd2 command like \binom{1}{2} => binom(1, 2)
+
+```typst
+#let cmd2 = ((kind: "cmd2"), none)
+```
+
+**Return:** A cmd2 spec and a scope item (none for no scope item)
+
+
+### `matrix-env`
+
+Define a matrix environment without handler
+
+```typst
+#let matrix-env = ((kind: "matrix-env"), none)
+```
+
+**Return:** A matrix-env spec and a scope item (none for no scope item)
+
+
+### `normal-env`
+
+Define a normal environment with handler
+
+```typst
+#let normal-env(handle) = ((kind: "normal-env"), (handle: handle))
+```
+
+**Arguments:**
+
+- handle (function): The handler function, as the value of alias in mitex-scope.
+  For example, define how to handle `aligned(..)` in mitex-scope
+
+**Return:** A normal-env spec and a scope item (none for no scope item)
+
+
+### `process-spec`
```

Comment:
not a function really related to command specification. It is a part of mitex.typ?